morango/api/viewsets.py

```diff
@@ -375,30 +377,59 @@ class BufferViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
 
     def create(self, request):
         data = request.data if isinstance(request.data, list) else [request.data]
-        serial_data = serializers.BufferSerializer(data=data, many=True)
-        if serial_data.is_valid():
-
-            # ensure the transfer session allows pushes, and is same across records
-            transfer_session = serial_data.validated_data[0]["transfer_session"]
-            if not transfer_session.push:
-                return response.Response(
-                    "Specified TransferSession does not allow pushes.",
-                    status=status.HTTP_403_FORBIDDEN
-                )
-            if len(set(rec["transfer_session"] for rec in serial_data.data)) > 1:
-                return response.Response(
-                    "All pushed records must be associated with the same TransferSession.",
-                    status=status.HTTP_403_FORBIDDEN
-                )
+        # ensure the transfer session allows pushes, and is same across records
+        transfer_session = TransferSession.objects.get(id=data[0]["transfer_session"])
+        if not transfer_session.push:
+            return response.Response(
+                "Specified TransferSession does not allow pushes.",
+                status=status.HTTP_403_FORBIDDEN
+            )
+        if len(set(rec["transfer_session"] for rec in data)) > 1:
+            return response.Response(
+                "All pushed records must be associated with the same TransferSession.",
+                status=status.HTTP_403_FORBIDDEN
+            )
 
-            serial_data.save()
+        rmcb_list = []
+        buffer_list = []
+        for record in data:
+            # ensure the provided model_uuid matches the expected/computed id
+            try:
+                Model = _profile_models[record["profile"]][record["model_name"]]
+            except KeyError:
+                Model = SyncableModel
+
+            partition = record['partition'].replace(record['model_uuid'], Model.ID_PLACEHOLDER)
+            expected_model_uuid = Model.compute_namespaced_id(partition, record["source_id"], record["model_name"])
+            if expected_model_uuid != record["model_uuid"]:
+                return response.Response({"model_uuid": "Does not match results of calling {}.compute_namespaced_id".format(Model.__class__.__name__)}, status=status.HTTP_400_BAD_REQUEST)
+
```

Comment:
Looks like there's an old version of the validation logic left in the serializer? https://github.com/learningequality/morango/blob/7853d2142dd266bfc06a8f673899549f8456c4e8/morango/api/serializers.py#L93

Perhaps good to clear all the unused stuff out over there to avoid confusion?